### ISSUE TITLE
LDK Node gateway backend with LSP Spec integration

### ISSUE THREAD

**Issue by elsirion** [fedimint/fedimint#4095]
Having a minimal LN node implementation that can only talk to an LSP could simplify setting up a LN gateway a lot. To be specced out.

**Comment by m1sterc001guy** [comment]
@tvolk131 for visibility

We already have some minimal LDK node support in our [fedimint-testing crate](https://github.com/fedimint/fedimint/blob/f391a86cf4d99f37079633bbddcda79abca45c7c/fedimint-testing/src/ln/real.rs#L372). It does not currently support receives.

**Comment by elsirion** [comment]
@tnull helped us better understand the capabilities of LDK [on a call yesterday](https://discord.com/channels/990354215060795454/1127969523203579944/1199026486011773008). 

One of the limitations of LDK is that it cannot intercept and claim random HTLCs, but rather expects incoming payments to be announced by giving it the payment hash and letting it generate the invoice.

This means that we cannot use our current protocol for LDK as-is, but need to adapt it:

* **User Client**: the client has to fetch the invoice from the gateway instead of generating it itself. This can happen in parallel to posting the incoming contract offer with the federation imo.
* **Gateway**: The gateway cannot use the SCID to determine the federation to route the payment to anymore. Instead there seems to be a way to include arbitrary data in invoices (as TLV fields) that is included in the final onion, so can be returned by LDK (Node).

The fact that the invoice would be generated by the gateway's LN node instead of the client means that users are leaking more private data than with other LN integrations, thus I don't see this new protocol as a replacement for the old one (although maybe a bit prettier) and rather as something we want to support in parallel to get an easy to run LN gateway option.

I see the following way forward:
* Add a flag to the gateway registration that indicates the type of gateway protocol supported, which the client uses to start the correct state machine.
* Warn users when they want to set a non-default invoice description in case they are using an LDK gateway or disallow this alltogether.
* Longer term work with LDK to make it compatible with our default protocol while improving it (better fake SCID choices as @tnull pointed us to, maybe more that I don't remember).

**Comment by elsirion** [comment]
Thx, totally forgot, I learnt a lot yesterday! Fixed up my post.

**Comment by justinmoon** [comment]
dev call: this could help with getting the https://github.com/fedimint/fedimint/pull/3468 running in a test environment

**Comment by tvolk131** [comment]
> One of the limitations of LDK is that it cannot intercept and claim random HTLCs

HTLC interception in LDK is tracked [here](https://github.com/lightningdevkit/rust-lightning/issues/2855) (although I don't believe it's currently being prioritized)

**Comment by m1sterc001guy** [comment]
I think this is completed! We can open a separate issue for LSP Spec integration :)
